src/snapred/ui/presenter/InitializeCalibrationPresenter.py

```diff
@@ -0,0 +1,119 @@
+import json
+
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import QLabel, QVBoxLayout, QWidget
+
+from snapred.backend.api.InterfaceController import InterfaceController
+from snapred.backend.dao.SNAPRequest import SNAPRequest
+from snapred.backend.dao.SNAPResponse import SNAPResponse
+from snapred.backend.dao.StateConfig import StateConfig
+from snapred.ui.threading.worker_pool import WorkerPool
+from snapred.ui.view.PromptUserforCalibrationInputView import PromptUserforCalibrationInputView
+
+
+class CalibrationCheck(object):
+    worker_pool = WorkerPool()
+    interfaceController = InterfaceController()
+
+    def __init__(self, view):
+        self.view = view
+
+    def _labelView(self, text: str):
+        win = QWidget()
+        vbox = QVBoxLayout()
+        label = QLabel(text)
+        label.setAlignment(Qt.AlignCenter)
+        vbox.addWidget(label)
+        vbox.addStretch()
+        win.setLayout(vbox)
+        self.view.layout().addWidget(win)
+
+    def handleButtonClicked(self):
+        self.view.beginFlowButton.setEnabled(False)
+
+        # Check to see if data exists
+        runNumber = self.view.getRunNumber()
+        runNumber_str = str(runNumber)
+        dataCheckRequest = SNAPRequest(
+            path="/calibration/checkDataExists", payload=json.dumps({"runNumber": runNumber_str})
+        )
+
+        self.worker = self.worker_pool.createWorker(
+            target=self.interfaceController.executeRequest, args=(dataCheckRequest)
+        )
+        self.worker.result.connect(self.handleDataCheckResult)
+
+        self.worker_pool.submitWorker(self.worker)
+
+    def handleDataCheckResult(self, response: SNAPResponse):
+        if response.responseCode != 200:
+            self._labelView("Error, data doesn't exist")
+            self.view.beginFlowButton.setEnabled(True)
+            return
+        else:
+            pass
+
+        runNumber_str = str(self.view.getRunNumber())
+        stateCheckRequest = SNAPRequest(path="/calibration/hasState", payload=json.dumps({"runNumber": runNumber_str}))
+
+        self.worker = self.worker_pool.createWorker(
+            target=self.interfaceController.executeRequest, args=(stateCheckRequest)
+        )
+        self.worker.result.connect(self.handleStateCheckResult)
+
+        self.worker_pool.submitWorker(self.worker)
+
+    def handleStateCheckResult(self, response: SNAPResponse):
+        if response.responseCode != 200:
+            self._spawnStateCreationWorkflow()
+            return
+        else:
+            pass
+
+        groupingFile = str(StateConfig.focusGroups.definition)  # This is incorrect, how do I find the groupingFile??
```

Comment:
I think this needs to be populated with a real value.  You may need to write up a separate method that looks up the focusGroups and feeds them to calculatePixelGroupingParameters.  You  can look them up by exposing[ this method](https://github.com/neutrons/SNAPRed/blob/next/src/snapred/backend/data/LocalDataService.py#L161) to the DataFactoryService.  So then you can just foreach over the groups.

Not saying you need to do something about this -> but, Realistically "/calibration/calculatePixelGroupingParameters" shouldnt be user facing as we wouldnt expect them to provide custom group files.

Reply:
Please reference edd5226 for fixes. Hopefully this works?